testRequirement.coffee

```diff
@@ -47,39 +52,64 @@ inRange = (objectKey, ranges, object) ->
 
   return valid
 
+# Check if the test is a substring test
 isSubstringTest = (testValues) ->
   if (_.isPlainObject testValues) and 'substring' of testValues then true else false
 
+# Perform the substring test, ignore case by default or if specified
 matchesSubstring = (objectKey, substringTest, object) ->
+
   substringValues = substringTest['substring'] # array of test substrings
-  objectValue = object[objectKey]
-  objectValues = if util.isArray objectValue then objectValue else [ objectValue ]
+  substringValues = if util.isArray substringValues then substringValues else [ substringValues ] #put substringValues in array if it isn't already
+
+  objectValue = object[objectKey] # array of document field values
+  objectValues = if util.isArray objectValue then objectValue else [ objectValue ] #put objectValues in array if it isn't already
+
   for objectValue in objectValues
     for substring in substringValues
+
+      if objectKey is 'reviews' # concatenate all reviews objects
```

Comment:
We can't do this :( This shouldn't know anything about the type of object it's testing, I'm gonna remove this before merging